src/rust/ensogl/src/math/geometry/triangle.rs

```diff
@@ -0,0 +1,169 @@
+/// return type within this module and should not otherwise be instantiated.
+///
+///
+/// Triangle Schematic
+/// -------------------
+/// ```text
+///                        C
+///                       / \
+///                     /    \
+///                   /       \
+///                 /  angle_c \
+///    side_ca    /             \  side_bc
+///             /                \
+///           /                   \
+///         /                      \
+///       /                         \
+///     /  angle_a           angle_b \
+///   /------------------------------- B
+///  A                 side_ab
+/// ```
+/// Where side_ab, side_bc and side_ca are the length of the respective side.
+/// Where alpha, beta and gamma are the respective angle.
+///
+/// Example
+/// -------
+/// ```
+/// # use assert_approx_eq::assert_approx_eq;
+/// # use ensogl::math::geometry::triangle::Triangle;
+/// let result = Triangle::from_sides_and_angle(1.0, 1.0 , 60.0_f32.to_radians());
+///
+/// assert_approx_eq!(result.side_bc(), 1.0);
+/// assert_approx_eq!(result.angle_a(), 60.0_f32.to_radians());
+/// assert_approx_eq!(result.angle_c(), 60.0_f32.to_radians());
+/// ```
+#[allow(missing_docs)]
+#[derive(Debug)]
+pub struct Triangle<T> {
+    side_bc: T,
+    side_ca: T,
+    side_ab: T,
+    angle_a: T,
+    angle_b: T,
+    angle_c: T,
+}
+
+#[allow(missing_docs)]
+impl<T> Triangle<T> {
+    pub fn side_bc(&self) -> &T {
+        &self.side_bc
+    }
+
+    pub fn side_ca(&self) -> &T {
+        &self.side_ca
+    }
+
+    pub fn side_ab(&self) -> &T {
+        &self.side_ab
+    }
+
+    pub fn angle_a(&self) -> &T {
+        &self.angle_a
+    }
+
+    pub fn angle_b(&self) -> &T {
+        &self.angle_b
+    }
+
+    pub fn angle_c(&self) -> &T {
+        &self.angle_c
+    }
+}
+
+pub trait TriangleInput<T> = Field<T>
+                           + Sin<Output=T>
+                           + Cos<Output=T>
+                           + Acos<Output=T>
+                           + Sqrt<Output=T>
+                           + Clone
+                           + From<f32>;
+
+impl<T> Triangle<T>
+where T:TriangleInput<T> {
+    /// Compute a triangle from two sides and the included angle (SAS).
+    /// See https://en.wikipedia.org/wiki/Solution_of_triangles#Two_sides_and_the_included_angle_given_(SAS)
+    pub fn from_sides_and_angle
+    (side_bc:T, side_ca:T, angle_c:T) -> Triangle<T> {
+        let side_bc_squared = side_bc.clone() * side_bc.clone();
+        let side_ca_squared = side_ca.clone() * side_ca.clone();
+
+        let two = T::from(2.0_f32);
+
+        let side_ab_squared_minuend    = side_bc_squared.clone() + side_ca_squared.clone();
+        let side_ab_squared_subtrahend = two.clone()
+                                         * side_bc.clone()
+                                         * side_ca.clone()
+                                         * angle_c.cos();
```

Comment:
This is not the most efficient impl here. Consider `Var<f32>` it could be string under the hood. Cloning LHS and RHS in order to compute result and throw LHS and RHS away is not optimal. Instead you could multiply references, like `&two * &side_bc`. In context of numbers, it would be the same, but for `Var`s it would actually be faster. After these changes the expressions should get shorter and fit single lines.

Reply:
The cloning is solved now, but the trait bounds are a bit uglier now.
The lines are still too long, even now.